engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -573,6 +573,43 @@ describe('System test: QuerySparql', () => {
         expect((await arrayifyStream(await result.execute()))).toEqual([]);
       });
     });
+
+    // TODO: better name here before merge.
+    describe('Comunica sparql regression test for #1029', () => {
+      it('Supports EXISTS sparql keyword', async() => {
+        const s = DF.namedNode('s');
+        const p = DF.namedNode('p');
+        const o = DF.namedNode('o');
+        const b = DF.blankNode();
+
+        const nodes = [ DF.quad(s, p, o), DF.quad(b, p, o) ];
+
+        const store = new Store(nodes);
+
+        const result = await engine.queryQuads(
+          `CONSTRUCT {?s ?p ?o.} WHERE {?s ?p ?o. OPTIONAL {FILTER EXISTS {?s2 ?p ?o. FILTER (?s != ?s2)}}}`,
+          { sources: [ store ]},
+        );
+        expect(await arrayifyStream(result)).toEqual([
+          {
```

Comment:
These should be made with a DataFactory.

Reply:
i was unable to get the tests to pass that way, but i'll give that more work! the blank nodes ended up with hard to predict literal names and that caused the equality tests to fail.

Reply:
We probably don't need the blank nodes? All named nodes should be fine.

Reply:
mk, switching it over.

Reply:
gonna push a failing version up so you can point out the fix; it's obviously a fairly trivial cause of failure given the size of the diff. my suspicion is that the sparql construct query in the test (also, full disclosure: i don't really know SPARQL!) and the data factory for named nodes just don't operate in the same way.

tl;dr: our fixture makes named nodes where the sparql query seems to make objects (??? not sure what that means in this context, but, it is the word the diff uses)

Reply:
See https://github.com/comunica/comunica/pull/1046/files#r1067792166